src/main.rs

```diff
@@ -62,35 +121,151 @@ fn main() {
+}
+
+fn row_slot_mut(table: &mut Table, row_num: u32) -> &mut [u8] {
+    let page_num = row_num / ROWS_PER_PAGE as u32;
+    let page = table.pages.entry(page_num).or_insert_with(|| Page {
+        buffer: [0u8; PAGE_SIZE],
+    });
+    let row_offset = row_num % ROWS_PER_PAGE as u32;
+    let byte_offset = row_offset * ROW_SIZE as u32;
+    &mut page.buffer[byte_offset as usize..byte_offset as usize + ROW_SIZE]
+}
+
+fn execute_statement<'a>(
+    statement: Statement<'a>,
+    table: &'a mut Table,
+) -> Result<ReplResult<'a>, ExecuteError> {
+    match statement {
+        Statement::Insert { row } => {
             println!("executing insert statement");
-            Ok(Results {})
+            let bytes = serialize_row(&row);
+            let mut point = row_slot_mut(table, table.num_rows);
+            point.write_all(&bytes).map_err(ExecuteError::Write)?;
+            table.num_rows += 1;
+            Ok(ReplResult::Success)
         }
-        StatementType::Select => {
+        Statement::Select => {
             println!("executing select statement");
-            Ok(Results {})
+            let mut rows = Vec::new();
+
+            for i in 0..table.num_rows {
+                let point = row_slot(table, i);
+                let sized_point = point.try_into().map_err(ExecuteError::RowRead)?;
+                let row = deserialize_row(sized_point);
+                rows.push(row);
+            }
+            Ok(ReplResult::Rows(rows))
         }
     }
 }
 
 fn prepare_statement(original_input: &str) -> Result<Statement, StatementError> {
     println!("processing statement {:?}", original_input);
-    let statement_type = if original_input.starts_with("insert") {
-        Ok(StatementType::Insert)
+    if original_input.starts_with("insert") {
+        let mut parts = original_input.split(' ');
+        let id = parts.nth(1);
+        let username = parts.next();
+        let email = parts.next();
+        match (id, username, email) {
+            (Some(id), Some(username), Some(email)) => {
+                let id = id.parse().map_err(|_| StatementError::Sql)?;
+
+                let username = username.as_bytes();
+                let email = email.as_bytes();
+
+                Ok(Statement::Insert {
+                    row: Row {
+                        id,
+                        username,
+                        email,
+                    },
+                })
+            }
+            _ => Err(StatementError::Sql),
+        }
     } else if original_input.starts_with("select") {
-        Ok(StatementType::Select)
+        Ok(Statement::Select)
     } else {
         Err(StatementError::Sql)
-    };
-    statement_type.map(|st| Statement { statement_type: st })
+    }
 }
 
 fn read_user_input(input_buffer: &mut String) -> Result<&str, ReplErr> {
     flush_stdout()
         .and_then(|_| stdin().read_line(input_buffer))
         .and_then(move |n| ensure_stdout_newline((n, input_buffer)))
+        .map(|s| s.trim())
```

Comment:
remove surrounding whitespace for easier work later on where whitespace around the input is irrelevant.